### PR TITLE
Feature/back/ar 10 create animal endpoint

### PR DESCRIPTION
**Pull request by AlexeyDerepa**

### CODE REVIEW

File: Backend/AnimalRescue/AnimalRescue.API/Controllers/AnimalController.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace AnimalRescue.API.Controllers
+{
+    public class AnimalController : ApiControllerBase
```
Comment (AlistratovAlex) [inline Backend/AnimalRescue/AnimalRescue.API/Controllers/AnimalController.cs]: AnimalController ->AnimalsController

Comment (AlexeyDerepa) [reply Backend/AnimalRescue/AnimalRescue.API/Controllers/AnimalController.cs]: done

File: Backend/AnimalRescue/AnimalRescue.API/Controllers/AnimalController.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        private readonly ILogger<HomeController> _logger;
+        public AnimalController(ILogger<HomeController> logger)
+        {
+            _logger = logger;
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<CollectionSegmentApiResponse<AnimalModel>>> Get()
+        {
+
```
Comment (golovnyov) [inline Backend/AnimalRescue/AnimalRescue.API/Controllers/AnimalController.cs]: Remove empty rows

File: Backend/AnimalRescue/AnimalRescue.API/Controllers/AnimalController.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlexeyDerepa) [reply Backend/AnimalRescue/AnimalRescue.API/Controllers/AnimalController.cs]: These attributes are in the ApiControllerBase

File: Backend/AnimalRescue/AnimalRescue.API/Controllers/AnimalController.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    public class AnimalController : ApiControllerBase
+    {
+        private readonly ILogger<HomeController> _logger;
+        public AnimalController(ILogger<HomeController> logger)
+        {
+            _logger = logger;
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<CollectionSegmentApiResponse<AnimalModel>>> Get()
```
[the remark this answers is not in the document]
Comment (AlexeyDerepa) [reply Backend/AnimalRescue/AnimalRescue.API/Controllers/AnimalController.cs]: done

File: Backend/AnimalRescue/AnimalRescue.API/Controllers/AnimalController.cs (hunk 2 above)
Comment (AlexeyDerepa) [reply Backend/AnimalRescue/AnimalRescue.API/Controllers/AnimalController.cs]: done

File: Backend/AnimalRescue/AnimalRescue.API/Controllers/HomeController.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+namespace AnimalRescue.API.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class HomeController : ControllerBase
```
[the remark this answers is not in the document]
Comment (AlexeyDerepa) [reply Backend/AnimalRescue/AnimalRescue.API/Controllers/HomeController.cs]: done

File: Backend/AnimalRescue/AnimalRescue.API/Startup.cs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddControllers();
+            
+            services.AddSConfigureSwagger();
```
[the remark this answers is not in the document]
Comment (AlexeyDerepa) [reply Backend/AnimalRescue/AnimalRescue.API/Startup.cs]: done

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/Models/Animal.cs [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace AnimalRescue.DataAccess.Mongodb.Models
+{
+    public class Animal : BaseItem
+    {    
+        public string ID { get; set; }
+        public long Number { get; set; }
```
[the remark this answers is not in the document]
Comment (AlexeyDerepa) [reply Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/Models/Animal.cs]: done

File: Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/Models/Animal.cs [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+using MongoDB.Bson.Serialization.Attributes;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace AnimalRescue.DataAccess.Mongodb.Models
+{
+    public class Animal : BaseItem
+    {    
+        public string ID { get; set; }
```
[the remark this answers is not in the document]
Comment (AlexeyDerepa) [reply Backend/AnimalRescue/AnimalRescue.DataAccess.Mongodb/Models/Animal.cs]: Id

File: Backend/AnimalRescue/AnimalRescue.Models.DTO/Models/AnimalModel.cs [hunk 8]
```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections.Generic;
+
+namespace AnimalRescue.Models.DTO.Models
+{
+    public class AnimalModel : BaseModel
+    {  
+        public long Number { get; set; }
```
[the remark this answers is not in the document]
Comment (AlexeyDerepa) [reply Backend/AnimalRescue/AnimalRescue.Models.DTO/Models/AnimalModel.cs]: done

File: Backend/AnimalRescue/AnimalRescue.API/Core/Middlewares/SwaggerExtension.cs [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace AnimalRescue.API.Core.Middlewares
+{
+    public static class SwaggerExtension
+    {
+        public static void AddSConfigureSwagger(this IServiceCollection services)
```
[the remark this answers is not in the document]
Comment (AlexeyDerepa) [reply Backend/AnimalRescue/AnimalRescue.API/Core/Middlewares/SwaggerExtension.cs]: done

File: Backend/AnimalRescue/AnimalRescue.API/Core/Middlewares/SwaggerExtension.cs [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+            // Register the Swagger generator, defining 1 or more Swagger documents  
+            services.AddSwaggerGen(c =>
+            {
+                c.SwaggerDoc(
+                    "v1",
+                    new OpenApiInfo
+                    {
+                        Title = "Animal Rescue API",
+                        Version = "v1",
+                        Description = "Animal Rescue API was build by Global Logic team"
```
[the remark this answers is not in the document]
Comment (AlexeyDerepa) [reply Backend/AnimalRescue/AnimalRescue.API/Core/Middlewares/SwaggerExtension.cs]: done

File: Backend/AnimalRescue/AnimalRescue.API/Controllers/AnimalController.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlistratovAlex) [reply Backend/AnimalRescue/AnimalRescue.API/Controllers/AnimalController.cs]: Thanks. I see

**Pull request merged**